services/datalake/pod-datalake/src/datalake/db.ts

```diff
@@ -284,6 +294,34 @@ export class PostgresDB implements BlobDB {
     )
   }
 
+  async getMeta (ctx: MeasureContext, blobId: BlobId): Promise<BlobMeta | null> {
+    const { workspace, name } = blobId
+
+    const rows = await this.db.execute<BlobMetaRecord[]>(
+      `
+      SELECT m.workspace, m.name, m.meta
+      FROM blob.blob b
+      LEFT JOIN blob.meta as m ON m.workspace = b.workspace AND m.name = b.name
+      WHERE b.workspace = $1 AND b.name = $2 and b.deleted_at IS NULL
```

Comment:
Do we need index for `deleted_at`?

Reply:
no, we don't need
it would be needed if we used sorting on delete date, but for nul//not null queries it will have low cardinality and won't be used

Reply:
Looks like the following index can be used in this case:
`CREATE INDEX ON blob (deleted_at) WHERE deleted_at IS NULL;`
But I don't mind leaving it as is

Reply:
Index on boolean field is usually a bad idea because it has low selectivity. Moreover, in this query we already filter by primary key (workspace and name) and deleted_at` is used as filter.